### PR TITLE
Event specific statuses

### PR DESCRIPTION
**Pull request by Joni-Aaltonen**
PR for issue #32

needs a bit more testing before deployment

database fields for statuses shown below, these will be migratet automatically upon deployment of the API

image

### CODE REVIEW

File: app/views/public-items.html [hunk 1]
```diff
@@ -127,17 +127,16 @@ <h2 class="panel-title">{{ category.category.name.text }}</h2>
...
-                              <a ng-if="item.status.id != ITEM_STATUS.GREEN" class="pull-right btn btn-danger btn-xs" ng-click="suggestDeletion(item)" style="margin-right:8px;"><span
-                              class="glyphicon glyphicon-remove"></span></a>
-                              <span ng-if="item.status.id == ITEM_STATUS.GREEN">&nbsp;</span>
+                              <span ng-if="item.status.allow_editing">
+                                <a class="pull-right btn btn-warning btn-xs" ng-click="openSuggestModalAside(parent, item)" style="margin-right:8px;"><span
+                                  class="glyphicon glyphicon-edit"></span></a>
+                                <a class="pull-right btn btn-danger btn-xs" ng-click="suggestDeletion(item)" style="margin-right:8px;"><span
+                                class="glyphicon glyphicon-remove"></span></a>
+                                <span>&nbsp;</span>
+                              </span>
```
Comment (fabian) [inline app/views/public-items.html]: This change causes some issues with the layout of the buttons, can you check what's wrong here?

screenshot 2018-05-08 14 36 49

File: app/scripts/controllers/skillcategory.js [hunk 2]
```diff
@@ -350,20 +349,20 @@ angular.module('ilApp')
...
 
     $scope.canEditStatus = function ()
     {
       return Auth.hasRole(APP_ROLES.EDIT_ITEM_STATUS);
     }
 
-    $scope.canDelete = function (statusId) {
-      return (statusId == ITEM_STATUS.RED) ? true : false;
+    $scope.canDelete = function (item) {
+      return item.status.allow_editing;
```
Comment (fabian) [inline app/scripts/controllers/skillcategory.js]: This is basically wrong, Organizer and Admins should always be able to delete. And then it's duplicated code with `canEdit`. Can you simply delete `canDelete` and replace it everywhere with `canEdit`?

**Comment by fabian** [comment]
Can you also rebase it on `master`? Sorry I made some small changes there related to statuses.

File: app/views/public-items.html (hunk 1 above)
Comment (Joni-Aaltonen) [reply app/views/public-items.html]: I didn't change the layout, it's exactly what it was earlier, but I can tweak it

File: app/scripts/controllers/skillcategory.js (hunk 2 above)
Comment (Joni-Aaltonen) [reply app/scripts/controllers/skillcategory.js]: canDelete wasn't actually used anywhere, got rid of it

**Comment by Joni-Aaltonen** [comment]
rebased from master and fixed @fabian

**Pull request merged**